### PR TITLE
PR for #768: Create tearsheet component

### PR DESCRIPTION
**Pull request by matthewgallo**
Contributes to #768 

This adds the `CreateTearsheet` component and a sub component, `CreateTearsheetStep`.

#### What did you change?
`CreateTearsheet/CreateTearsheet.js`
`CreateTearsheet/CreateTearsheet.mdx`
`CreateTearsheet/CreateTearsheet.stories.js`
`CreateTearsheet/CreateTearsheet.test.js`
`CreateTearsheet/CreateTearsheetStep.js`
`CreateTearsheet/_create-tearsheet.scss`
`CreateTearsheet/_index.scss`
`CreateTearsheet/_storybook-styles.scss`
`CreateTearsheet/constants.js`
`CreateTearsheet/index.js`
`CreateTearsheet/preview-components/MultiStepTearsheet.js`
`packages/cloud-cognitive/src/components/_index.scss`
`packages/cloud-cognitive/src/components/index.js`
`package-settings.js`
#### How did you test and verify your work?
Storybook and `jest /CreateTearsheet/ --coverage`

*Note: @sydrosa and @dcwarwick I removed the ability to have custom next button text per step, as well as adding the functionality of focusing the first available input on a step change (if there isn't a focusable element in a step, the focus is set on the Next/Create button), and also not requiring `onNext` and `onRequestSubmit` to return promises, this is optional now.

### CODE REVIEW

**Comment by davidmenendez** [comment]
what are `preview-components`?

**Comment by matthewgallo** [comment]
Those are just components that are used in Storybook

File: packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.mdx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Example usage](#example-usage)
+- [Component API](#component-api)
+
+## Overview
+
+This component is an example
```
Comment (lee-chase) [inline packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.mdx:16]: Is it an example?

File: packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.test.js [hunk 2]
```diff
@@ -0,0 +1,319 @@
+/* eslint-disable react/prop-types */
```
Comment (lee-chase) [inline packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.test.js:1]: No ref or additional prop test.

File: packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheetStep.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ */
+
+import React, { forwardRef } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { pkg } from '../../settings';
+import { CREATE_TEARSHEET_STEP } from './constants';
+
+const componentName = 'CreateTearsheetStep';
+const blockClass = `${pkg.prefix}--tearsheet__step`;
```
Comment (lee-chase) [inline packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheetStep.js:15]: Should the block class not reference 'create tearsheet'?

Comment (lee-chase) [reply packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheetStep.js:15]: Does this component need to accept additional props?

File: packages/cloud-cognitive/src/components/CreateTearsheet/_create-tearsheet.scss [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+// LICENSE file in the root directory of this source tree.
+//
+
+@import '@carbon/import-once/scss/import-once';
+@import '../../global/styles/carbon-settings';
+@import '../../global/styles/project-settings';
+@import 'carbon-components/scss/components/progress-indicator/progress-indicator';
+
+@mixin create-tearsheet {
+  $block-class: #{$pkg-prefix}--tearsheet;
```
Comment (lee-chase) [inline packages/cloud-cognitive/src/components/CreateTearsheet/_create-tearsheet.scss:14]: Should this be 'create-tearsheet'?

File: packages/cloud-cognitive/src/components/CreateTearsheet/preview-components/MultiStepTearsheet.js [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+  TextInput,
+  Toggle,
+  NumberInput,
+} from 'carbon-components-react';
+import cx from 'classnames';
+import { pkg } from '../../../settings';
+import { CreateTearsheet } from '../CreateTearsheet';
+import { CreateTearsheetStep } from '../CreateTearsheetStep';
+
+const blockClass = `${pkg.prefix}--tearsheet-wizard-preview`;
```
Comment (lee-chase) [inline packages/cloud-cognitive/src/components/CreateTearsheet/preview-components/MultiStepTearsheet.js:23]: Should this reference 'multi step tearsheet'?

**Review by lee-chase — COMMENTED** [review]
See comments

File: packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.mdx (hunk 1 above)
Comment (matthewgallo) [reply packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheet.mdx:16]: Yeah I need to update the doc file here 👍

File: packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheetStep.js (hunk 3 above)
Comment (matthewgallo) [reply packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheetStep.js:15]: I don't think it need additional props, this is simply a wrapper component for each step. It does have props of it's own but not likely that it needs additional. And yes, I agree the block class should reflect the create tearsheet block class, something like this `${pkg.prefix}--tearsheet-create__step`

**Comment by matthewgallo** [comment]
I've updated the `blockClass` naming in some of the components and added the missing tests in the [latest commit](https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/785/commits/ab717871b60cea77129bdec7ad11f9a4ca02c92a) @lee-chase

**Review by lee-chase — APPROVED** [review]

**Pull request merged**